docs/ecosystem/oracles/pragma.md

```diff
@@ -1,3 +1,147 @@
+        uint80 answeredInRound
+    );
+}
+```
+
+#### 2. Retrieve the BTC/USD Spot Median Price
+
+```typescript
+contract PragmaDataConsumer {
+    IPragmaAggregatorV3 internal dataFeed;
+
+    constructor() {
+        dataFeed = IPragmaAggregatorV3(
+            0x5a3d161e5c63511F97F51fbF366B8238Cd0bBeAc // BTC/USD Pragma Interface
+        );
+    }
+
+    function getPragmaDataFeedLatestPrice() public view returns (int256) {
+        (
+            /* uint80 roundID */,
+            int256 price,
+            /*uint256 startedAt*/,
+            /*uint256 timeStamp*/,
+            /*uint80 answeredInRound*/
+        ) = dataFeed.latestRoundData();
+        return price;
+    }
+}
+```
+
+## 2. Directly call the `PragmaCaller`
+
+The Pragma Oracle cairo contract has been integrated by Kakarot - thus it is possible to call it directly in your Solidity contracts using the [PragmaCaller](https://github.com/kkrt-labs/kakarot/blob/main/solidity_contracts%2Fsrc%2FCairoPrecompiles%2FPragmaCaller.sol) interface!
+
+The complete list of [supported assets](https://docs.pragma.build/Resources/Cairo%201/data-feeds/supported-assets) can be found in our documentation: just grab the pair id and you're good to go!
+
+#### 1. Define the PragmaCaller interface
+
+```typescript
+interface IPragmaCaller {
+    enum DataType {
+        SpotEntry,
+        FuturesEntry,
+        GenericEntry
+    }
+
+    struct DataRequest {
+        DataType dataType;
+        uint256 pairId;
+        uint256 expirationTimestamp;
+    }
+
+    struct PragmaPricesResponse {
+        uint256 price;
+        uint256 decimals;
+        uint256 last_updated_timestamp;
+        uint256 num_sources_aggregated;
+        uint256 maybe_expiration_timestamp;
+    }
+
+    function getDataMedianSpot(DataRequest memory request) external view returns (PragmaPricesResponse memory);
+}
+```
+
+#### 2. Retrieve the BTC/USD Spot Median Price
+
+```typescript
+interface IPragmaCaller {
+    // ... (include the interface definition from step 1)
+}
+
+contract CallerExample {
+    IPragmaCaller private pragmaCaller;
+    // ascii conversion of "BTC/USD";
+    uint256 constant BTC_USD_FEED = 18669995996566340;
+
+    constructor(address pragmaCallerAddress) {
+        pragmaCaller = IPragmaCaller(pragmaCallerAddress);
+    }
+
+    function getDataMedianSpot(uint256 pairId) public view returns (IPragmaCaller.PragmaPricesResponse memory) {
+        IPragmaCaller.DataRequest memory request = IPragmaCaller.DataRequest(
+            IPragmaCaller.DataType.SpotEntry,
+            pairId,
+            0
+        );
+        return pragmaCaller.getDataMedianSpot(request);
+    }
+
+    function getBtcMedianPrice() public view returns (uint256) {
+        IPragmaCaller.PragmaPricesResponse memory response = getDataMedianSpot(BTC_USD_FEED);
+        return response.price;
+    }
+}
+```
+
+An example has been deployed [here](https://sepolia.kakarotscan.org/address/0x9acb5dbE6B9E3569f4Ab8b4bf8E09F7efC330A26).
```

Comment:
Can you verify the contract so that they can see the code directly on the explorer? We can help with that

Reply:
Maybe adding a small github repository can go a long way too! People love to check out a "examples" repo

Reply:
> Can you verify the contract so that they can see the code directly on the explorer? We can help with that

All contracts are actually verified but since yesterday I can no longer access the "Contract" tab on the explorer - not sure why

Reply:
Ah weird